spring-cloud-stream-module-launcher/src/main/java/org/springframework/boot/loader/ModuleJarLauncher.java

```diff
@@ -80,12 +81,15 @@ protected void launch(String[] args, String mainClass, ClassLoader classLoader)
 	@Override
 	protected ClassLoader createClassLoader(URL[] urls) throws Exception {
 		ClassLoader systemClassLoader = ClassLoader.getSystemClassLoader();
-		// try to retrieve the extension classloader
-		ClassLoader extensionClassLoader = systemClassLoader != null ? systemClassLoader.getParent() : null;
-		// set the classloader for the module as the extension classloader if available
-		// fall back to the system classloader (which can also be null) if not available
-		ClassLoader classLoaderForModule = extensionClassLoader != null ? extensionClassLoader : systemClassLoader;
-		return new LaunchedURLClassLoader(urls, classLoaderForModule);
+		if (systemClassLoader instanceof URLClassLoader) {
+			URLClassLoader classLoader = (URLClassLoader) systemClassLoader;
+			URL[] toUse = new URL[urls.length + classLoader.getURLs().length];
+			System.arraycopy(urls, 0, toUse, 0, urls.length);
+			System.arraycopy(classLoader.getURLs(), 0, toUse, urls.length, classLoader.getURLs().length);
+			return new LaunchedURLClassLoader(toUse, null);
```

Comment:
Just did a quick test (dumping the added URLs) and this definitely skips extensions from the launched module's class loader. So we either need to set the extension class loader as parent here, or to copy the extension class loader URLs. I would be happier with the former, as it supersedes the behaviour of the JVM in the least (and I don't think we need to extend resource loading to that class loader).